### PR TITLE
Replace the HTTP transport Client field with unexported transport field.

### PR DESCRIPTION
**Pull request by ian-mi**
Add a new HTTP option which allows setting the HTTP transport while
keeping the field unexported for improved safety. This change will allow
the internal HTTP client to be constructed with additional middleware
if needed, for example, for trace propagation.

### CODE REVIEW

File: pkg/cloudevents/transport/http/transport.go [hunk 1]
```diff
@@ -44,11 +44,6 @@ type Transport struct {
 	// If nil, DefaultShutdownTimeout is used.
 	ShutdownTimeout *time.Duration
 
-	// Sending
-
-	// Client is the http client that will be used to send requests.
-	// If nil, the Transport will create a one.
-	Client *http.Client
```
Comment (n3wscott) [inline pkg/cloudevents/transport/http/transport.go]: This would be a breaking semver change, can you rework the PR to not change the api?

**Comment by n3wscott** [comment]
LGTM, but I don't want to bump the api to 2.0 yet, can you leave httptransport.Client in place and either mark it deprecated and/or use it with the new inner transport from context?

**Comment by ian-mi** [comment]
I've marked the Client field deprecated with the behavior that setting Client will cause WithHTTPTransport to be ignored.

File: pkg/cloudevents/transport/http/transport.go (hunk 1 above)
Comment (ian-mi) [reply pkg/cloudevents/transport/http/transport.go]: Done.

File: pkg/cloudevents/transport/http/transport.go [hunk 2]
```diff
@@ -46,9 +46,10 @@ type Transport struct {
 
 	// Sending
 
-	// Client is the http client that will be used to send requests.
-	// If nil, the Transport will create a one.
+	// Deprecated - setting http client will override use of the
```
Comment (n3wscott) [inline pkg/cloudevents/transport/http/transport.go]: THANKS!!!

**Comment by n3wscott** [comment]
Thanks for the non-breaking change to the API.

LGTM.

**Pull request merged**